x-pack/plugins/apm/scripts/kibana-security/create_kibana_user_role.ts

```diff
@@ -0,0 +1,534 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import axios, { AxiosRequestConfig, AxiosError } from 'axios';
+import { union, difference, once } from 'lodash';
+
+/* eslint-disable no-console */
+
+interface User {
+  username: string;
+  roles: string[];
+  full_name?: string;
+  email?: string;
+  enabled?: boolean;
+}
+
+export async function createKibanaUserRole({
+  kibanaRoleSuffix,
+  esUserName,
+  esPassword,
+  kibanaBaseUrl,
+}: {
+  esUserName: string;
+  kibanaRoleSuffix?: string;
+  esPassword?: string;
+  kibanaBaseUrl?: string;
+}) {
+  if (!esPassword) {
+    console.log(
+      'Please specify credentials for elasticsearch: `--username elastic --password abcd` '
+    );
+    return;
+  }
+
+  if (!kibanaBaseUrl) {
+    console.log(
+      'Please specify the url for Kibana: `--kibana-url http://localhost:5601` '
+    );
+    return;
+  }
+
+  if (
+    !kibanaBaseUrl.startsWith('https://') &&
+    !kibanaBaseUrl.startsWith('http://')
+  ) {
+    console.log(
+      'Kibana url must be prefixed with http(s):// `--kibana-url http://localhost:5601`'
+    );
+    return;
+  }
+
+  if (!kibanaRoleSuffix) {
+    console.log(
+      'Please specify a unique suffix that will be added to your roles with `--role-suffix <suffix>` '
+    );
+    return;
+  }
+
+  const version = await getKibanaVersion({
+    esPassword,
+    esUserName,
+    kibanaBaseUrl,
+  });
+  console.log(`Connected to Kibana ${version}`);
+
+  const isEnabled = await isSecurityEnabled({
+    esUserName,
+    esPassword,
+    kibanaBaseUrl,
+  });
+  if (!isEnabled) {
+    console.log('Security must be enabled!');
+    return;
+  }
```

Comment:
Since this is no longer just a cli script it should throw

```suggestion
  if (!isEnabled) {
    throw new Error('Security is not enabled');
  }
```